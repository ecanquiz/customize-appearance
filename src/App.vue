<script setup lang="ts">
import StylePanel from './components/StylePanel.vue';
import PreviewLayout from './components/PreviewLayout.vue';
import useApp from './composables/useApp';

const {
  currentStyles,

  handleStyleUpdate
} = useApp();
</script>

<template>
  <div class="app-container">
    <StylePanel 
      :styles="currentStyles" 
      @update-styles="handleStyleUpdate"
    />
    <PreviewLayout :styles="currentStyles" />
  </div>
</template>

<style>
@import '@/assets/styles/app.css';
</style>

