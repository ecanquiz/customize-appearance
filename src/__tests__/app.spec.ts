import { mount } from '@vue/test-utils'
import App from '@/App.vue'
import { describe, it, expect } from 'vitest'

describe('App.vue', () => {
  it('matches snapshot', () => {
    const wrapper = mount(App)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
