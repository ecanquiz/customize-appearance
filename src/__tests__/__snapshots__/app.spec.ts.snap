// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App.vue > matches snapshot 1`] = `
"<div class="app-container">
  <div data-v-8a782076="" class="style-panel">
    <h2 data-v-8a782076="">Personalización de estilos</h2>
    <div data-v-8a782076="" class="form-group"><label data-v-8a782076="">Color de fondo:</label><input data-v-8a782076="" type="color"><input data-v-8a782076="" type="text"></div>
    <div data-v-8a782076="" class="form-group"><label data-v-8a782076="">Color de texto:</label><input data-v-8a782076="" type="color"><input data-v-8a782076="" type="text"></div>
    <div data-v-8a782076="" class="form-group"><label data-v-8a782076="">Color principal:</label><input data-v-8a782076="" type="color"><input data-v-8a782076="" type="text"></div>
    <div data-v-8a782076="" class="form-group"><label data-v-8a782076="">Color de títulos:</label><input data-v-8a782076="" type="color"><input data-v-8a782076="" type="text"></div>
    <div data-v-8a782076="" class="form-group"><label data-v-8a782076="">Fondo del header:</label><input data-v-8a782076="" type="color"><input data-v-8a782076="" type="text"></div>
    <div data-v-8a782076="" class="form-group"><label data-v-8a782076="">Fondo del footer:</label><input data-v-8a782076="" type="color"><input data-v-8a782076="" type="text"></div><button data-v-8a782076="">Restablecer valores por defecto</button>
  </div>
  <div data-v-2578c25a="" class="preview-layout" style="background-color: rgb(255, 255, 255); color: rgb(51, 51, 51);">
    <header data-v-2578c25a="" class="preview-header" style="background-color: rgb(248, 249, 250);">
      <h1 data-v-2578c25a="" style="color: rgb(66, 185, 131);">Mi Sitio Web</h1>
      <nav data-v-2578c25a=""><a data-v-2578c25a="" href="#" style="color: rgb(66, 185, 131);">Inicio</a><a data-v-2578c25a="" href="#" style="color: rgb(66, 185, 131);">Acerca de</a><a data-v-2578c25a="" href="#" style="color: rgb(66, 185, 131);">Contacto</a></nav>
    </header>
    <main data-v-2578c25a="" class="preview-content">
      <article data-v-2578c25a="">
        <h2 data-v-2578c25a="" style="color: rgb(44, 62, 80);">Artículo Principal</h2>
        <p data-v-2578c25a="">Este es un ejemplo de contenido que puede ser personalizado según las preferencias del usuario.</p><button data-v-2578c25a="" class="primary-button" style="background-color: rgb(66, 185, 131); color: rgb(0, 0, 0);"> Botón de ejemplo </button>
      </article>
      <aside data-v-2578c25a="">
        <h3 data-v-2578c25a="" style="color: rgb(44, 62, 80);">Barra lateral</h3>
        <p data-v-2578c25a="">Contenido adicional que complementa el artículo principal.</p>
      </aside>
    </main>
    <footer data-v-2578c25a="" class="preview-footer" style="background-color: rgb(248, 249, 250);">
      <p data-v-2578c25a="" style="color: rgb(66, 185, 131);">© 2025 Mi Sitio Web</p>
    </footer>
  </div>
</div>"
`;
